### DISCUSSION TITLE
DiscordAPIError: Missing Access, Even though bot has been given admin

### DISCUSSION THREAD

**Discussion by WilliamHarer** [discordjs/voice#193] [Q&A] [answered]
Like the title says, been pulling my hair out dealing with my main bot getting stuck at signaling when starting a voiceConnection; so I jump over here and start pulling my hair out because even though my bot should have admin rights it's missing access.

**Comment by megatank58** [comment, marked as answer]
You also need application.commands scope while inviting the bot if the error is from slash commands/interactions

**Reply by WilliamHarer** [reply]
It should have that, I'm really not sure why it's mad.

![image](https://user-images.githubusercontent.com/46909464/131087917-487cbfec-947f-4066-b9c4-a2b1c253c0ab.png)

**Reply by megatank58** [reply]
Those are permissions for the the bot

To get application.commands scope

• Invite your bot with a link including the required scopes as explained here:  https://discordjs.guide/preparations/adding-your-bot-to-servers.html

**Reply by WilliamHarer** [reply]
That's unfortunately not the issue, this screen is in client and was pulled after the bot was added to the server.

**Reply by megatank58** [reply]
> That's unfortunately not the issue, this screen is in client and was pulled after the bot was added to the server.

- That is the issue though the error generally originates from slash command deploys without application.commands scope

**Reply by WilliamHarer** [reply]
I'm not sure where in the bot creation process I'm missing steps though :/

![image](https://user-images.githubusercontent.com/46909464/131089830-c6c6634b-4f13-41dd-87dd-d159d335d79e.png)
![image](https://user-images.githubusercontent.com/46909464/131089884-f78f8709-609d-4765-9058-f7286bcc52f6.png)
![image](https://user-images.githubusercontent.com/46909464/131090000-03f3d775-05a2-498b-b510-8ef1133ff097.png)
![image](https://user-images.githubusercontent.com/46909464/131090049-c015e6ba-8f7a-4c96-81a0-d1fac1029f31.png)
![image](https://user-images.githubusercontent.com/46909464/131090089-a6cf6304-a34a-446f-9626-35807eb696a4.png)
The only remaining step would be to change my auth.json token to the new bot token, which I do only to continue to receive the access error

**Reply by WilliamHarer** [reply]
Ah I caught it. The discord application webpage doesn't update the scope in the autogenerated url to add slash commands. Someone should look into updating the application auto-generating url to add that as a scope option.

**Reply by megatank58** [reply]
well its not auto generated, bot is also a scope and so is application.commands you must specifically choose them

**Reply by WilliamHarer** [reply]
I see it now, bot plays nicely with some url changing options in scope and not others. I guess I had to many selected so it was just telling me to choose a url instead of giving me the generated one. Thanks for the help

**Comment by amishshah** [comment]
Can you share some code?

**Reply by WilliamHarer** [reply]
It's purely cloned from the repository, no additions of my own.

Here's the error, the stacktrace, and the file location within the repo.

~~~~~
C:\Users\liamh\IdeaProjects\voice\examples\music-bot\node_modules\discord.js\src\rest\RequestHandler.js:298
      throw new DiscordAPIError(data, res.status, request);
            ^

DiscordAPIError: Missing Access
    at RequestHandler.execute (C:\Users\liamh\IdeaProjects\voice\examples\music-bot\node_modules\discord.js\src\rest\RequestHandler.js:298:13)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async RequestHandler.push (C:\Users\liamh\IdeaProjects\voice\examples\music-bot\node_modules\discord.js\src\rest\RequestHandler.js:50:14)
    at async GuildApplicationCommandManager.set (C:\Users\liamh\IdeaProjects\voice\examples\music-bot\node_modules\discord.js\src\managers\ApplicationCommandManager.js:146:18)
    at async Client.<anonymous> (C:\Users\liamh\IdeaProjects\voice\examples\music-bot\dist\bot.js:38:9) {
~~~~
